vm/map.c

```diff
@@ -719,6 +730,71 @@ int vm_munmap(vm_map_t *map, void *vaddr, size_t size)
 }
 
 
+int vm_mprotect(vm_map_t *map, void *vaddr, size_t len, int prot)
+{
+	int result = EOK;
+	size_t size;
+	addr_t pa;
+	int attrs = 0;
+	map_entry_t *e;
+
+	if ((((ptr_t)vaddr) & (SIZE_PAGE - 1)) != 0) {
+		return -EINVAL;
+	}
+
+	if ((prot & PROT_READ) != 0) {
+		attrs |= PGHD_PRESENT;
+	}
```

Comment:
Now that the GR712RC port is merged, please add `PGHD_READ` to `attrs` - on this target pages can be exec-only - https://github.com/phoenix-rtos/phoenix-rtos-kernel/commit/db4f5e5e7db0eb10af6e2386cee98b5c0dc89ed3